persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/UserDAOImpl.kt

```diff
@@ -179,6 +179,18 @@ class UserDAOImpl(
             .map { it?.let { mapper.toModel(it) } }
     }
 
+    override fun getUserMinimizedByQualifiedID(qualifiedID: QualifiedIDEntity): UserEntityMinimized? =
+        userQueries.selectMinimizedByQualifiedId(listOf(qualifiedID)).executeAsOneOrNull().let {
+            it?.let { user ->
+                UserEntityMinimized(
+                    user.qualified_id,
+                    user.name,
+                    user.complete_asset_id,
+                    user.user_type
+                )
+            }
+        }
```

Comment:
Instead of mapping after the query, it's more performant if we pass a mapper function as the last argument of `selectMinimizedByQualifiedId`:

```suggestion
        userQueries.selectMinimizedByQualifiedId(listOf(qualifiedID)) { 
            qualifiedId, name, completeAssetId, userType ->
                UserEntityMinimized(
                    qualifiedId,
                    name,
                    completeAssetId,
                    userType 
                )
        }.executeAsOneOrNull()
```

It's probably better to move this mapping function to `UserMapper` and just pass it like this:

```kotlin
        userQueries.selectMinimizedByQualifiedId(
            listOf(qualifiedID), userMapper::toMinimizedEntity
        ).executeAsOneOrNull()
```

Reply:
done